modes/vi-mode/core.lisp

```diff
@@ -55,27 +59,50 @@
   (setf (element-name *modeline-element*) name))
 
 
-(defstruct vi-state
-  name
-  keymap
-  post-command-hook
-  enable-hook
-  disable-hook
-  cursor-color)
+(defclass vi-state ()
+  ((tag
+  :initarg :tag
+  :reader tag)
+  (message
+  :initarg :message
+  :reader msg)
+  (cursor-type 
+  :initarg :cursor-type
+  :reader cursor-type)
+  (keymap
+  :initarg :keymap
+  :reader state-keymap)
+  (cursor-color
+  :initarg :cursor-color
+  :accessor cursor-color)))
```

Comment:
I think it is very good that it is now class :+1:

Reply:
I have come to think that the reader/accessor name is a good style to use prefix as class after seeing the following text, what do you think?
https://google.github.io/styleguide/lispguide.xml?showone=CLOS#CLOS